packages/gatsby-dev-cli/src/utils/check-deps-changes.js

```diff
@@ -61,15 +61,13 @@ exports.checkDepsChanges = async ({
     // and save some time/work
     try {
       localPKGjson = await new Promise((resolve, reject) => {
-        got(`https://unpkg.com/${packageName}/package.json`).then(
-          (error, response) => {
+        got(`https://unpkg.com/${packageName}/package.json`)
+          .then((error, response) => {
             if (response && response.statusCode === 200) {
-              return resolve(JSON.parse(response.body))
+              resolve(JSON.parse(response.body))
             }
-
-            return reject(error)
-          }
-        )
+          })
+          .catch(e => reject(e))
```

Comment:
`.then` handler can only have single argument - so this should be:

```
got(`https://unpkg.com/${packageName}/package.json`)
  .then(response => {
    if (response && response.statusCode === 200) {
      return resolve(JSON.parse(response.body))
    }

    return reject(new Error(`No response or non 200 code`))
  })
  .catch(e => reject(e))
```

Seems like migration in https://github.com/gatsbyjs/gatsby/pull/26653 was never actually tested?

Reply:
Considering this is inside an `async` function it should be something like

```js
const response = await got(`https://unpkg.com/${packageName}/package.json`)
if (response?.statusCode !== 200) {
  throw new Error(`No response or non 200 code`)
}
resolve(JSON.parse(response.body))
```

Reply:
(unless `got` does not return a proper promise-like?)

Reply:
Ah, yeah, you are probably right and `await` will simplify things - I mostly zoomed in on double args for `.then` :)

Reply:
>Seems like migration in #26653 was never actually tested?

Yeah — this isn't a common scenario.

Will change to await 👍